### DISCUSSION TITLE
Lucence query: howto build query to search from content picker field or predefined text field

### DISCUSSION THREAD

**Discussion by thanhpt168** [OrchardCMS/OrchardCore#7254] [Q&A] [answered]
I defined 2 content types.
1. ContenType: WorkingPlace with Title field
2. ContentType: MyJob, contain Cotent Picker Field which point to WorkingPlace type, and a predefined list text field,
Now, I want to build Lucence Query to search MyJob items by  WorkingPlace, and other query to search MyJob item by "value" of selected value from predefined list.  How can I build the query? I tried write the correct syntax but no luck.

My query to search MyJob items by WorkingPlace content picker field
```
{
  "query": {
    "bool": {
      "must": [
        {
          "term": {
            "Content.ContentItem.MyJob.MyWorkingPlace": "47374cffdd8zf2aknx42ajrp5p"
          }
        },
        {
          "term": {
            "Content.ContentItem.ContentType": "MyJob"
          }
        }
      ]
    }
  }
}
```

in which: - **MyWorkingPlace** is field name to WorkingPlace, **47374cffdd8zf2aknx42ajrp5p** is contentItemId of WorkingPlace.

Many thanks!

**Comment by Skrypt** [comment, marked as answer]
Have you set your MyWorkingPlace field as a "Stored" indexed field? Here it is a technical value so it should not be "Analyzed" aka tokenized.

**Reply by thanhpt168** [reply]
Yes, I did. Could you give me a sample query in case search by indexed content picker field?

**Reply by Skrypt** [reply]
Here is something that should definitely work :  

```liquid
{
  "from" : {{ from }}, "size" : {{ size }},
  "query": {
    "bool" : {
      "must" : [
        { "term" : { "Content.ContentItem.Published" : "true" }},
        {%if fulltext != null %}
        { "match" : { "Content.ContentItem.FullText" : "{{ fulltext }}" }},
        {%endif%}
        {
          "bool": {
            "should": [
              { "term" : { "Content.ContentItem.ContentType" : { "value": "YourContentType", "boost": 5 }}},
              { "wildcard" : { "YourContentType.Category": "{{ category }}"}}
            ]
          }
        },
        {
          "bool": {
            "should": [
              { "wildcard" : { "YourContentType.Brand": "{{ category }}"}}
            ]
          }
        },
        {% if owner != '*' %}
        { "term" : { "Content.ContentItem.Owner" : "{{ owner }}" } }
        {% endif %}
      ]
    }
  },
  "sort": [
    { "Content.ContentItem.ModifiedUtc": { "order": "desc" } },
    { "Content.ContentItem.CreatedUtc": { "order": "desc" } }
  ]
}
```

**Reply by thanhpt168** [reply]
I found the problem: I used wrong path. It must be MyJob.MyWorkingPlace instead of Content.ContentItem.MyJob.MyWorkingPlace.
The correct query is: 
```
{
  "query": {
    "bool": {
      "must": [
        {
          "term": {
            "MyJob.MyWorkingPlace": "47374cffdd8zf2aknx42ajrp5p"
          }
        },
        {
          "term": {
            "Content.ContentItem.ContentType": "MyJob"
          }
        }
      ]
    }
  }
}
```

Thank alot for your help!
